src/common/toggle.js

```diff
@@ -63,118 +55,132 @@ export const init =
     ]
-  : action.type === "Uncheck"
-  ? [ merge(model, {isChecked: false })
-    , Effects.none
+  : [ new Model(true, model.button)
+    , Effects.perform(Task.succeed(Check))
     ]
-  : action.type === "Button"
-  ? mapFX(ButtonAction, Button.update(model, action.button))
-  : action.type === "Target"
-  ? mapFX(TargetAction, Target.update(model, action.target))
-  : action.type === "Focusable"
-  ? mapFX(FocusableAction, Focusable.update(model, action.focusable))
-  : Unknown.update(model, action)
-  );
-
-
-export function view(key:string,
-                     styleSheet:StyleSheet):(model:Model, address:Address<Action>, contextStyle?:ContextStyle) => DOM {
-  return ( model
-         , address
-         , contextStyle
-  ):DOM =>
+  )
+
+export const press = toggle
+
+export const check =
+  (model:Model):[Model, Effects<Action>] =>
+  ( model.isChecked
+  ? nofx(model)
+  : nofx(new Model(true, model.button))
+  )
+
+export const uncheck =
+  (model:Model):[Model, Effects<Action>] =>
+  ( model.isChecked
+  ? nofx(new Model(false, model.button))
+  : nofx(model)
+  )
+
+const delegateButtonUpdate =
+  (model, action) =>
+  swapButton(model, Button.update(model.button, action))
+
+const swapButton =
+  (model, [button, fx]) =>
+  [ new Model(model.isChecked, button)
+  , fx.map(ButtonAction)
+  ]
+
+
+export const view =
+  (key:string, styleSheet:StyleSheet) =>
+  (model:Model, address:Address<Action>, contextStyle?:ContextStyle):DOM =>
   html.button({
     key: key,
     className: key,
     style: Style
       ( styleSheet.base
-
-      ,   model.isFocused
+      , ( model.isFocused
         ? styleSheet.focused
         : styleSheet.blured
-
-      ,   model.isDisabled
+        )
+      , ( model.isDisabled
         ? styleSheet.disabled
         : styleSheet.enabled
-
-
-      ,  model.isPointerOver
+        )
+      , ( model.isPointerOver
         ? styleSheet.over
         : styleSheet.out
-
-      ,  model.isActive
+        )
+      , ( model.isActive
         ? styleSheet.active
         : styleSheet.inactive
-
-      ,   model.isChecked
+        )
+      , ( model.isChecked
         ? styleSheet.checked
         : styleSheet.unchecked
-
+        )
       , contextStyle
-    ),
+      ),
 
-    onFocus: forward(address, always(Focus)),
+    onFocus: forward(address, always(Activate)),
     onBlur: forward(address, always(Blur)),
-
     onMouseOver: forward(address, always(Over)),
     onMouseOut: forward(address, always(Out)),
-
+    onClick: forward(address, always(Click)),
     onMouseDown: forward(address, always(Down)),
-    onMouseUp: forward(address, always(Up)),
-
-    onClick: forward(address, always(Press))
+    onMouseUp: forward(address, always(Up))
   });
-}
+
+export const Activate = ButtonAction(Button.Focus)
+export const Blur = ButtonAction(Button.Blur)
+export const Over = ButtonAction(Button.Over)
+export const Out = ButtonAction(Button.Out)
+export const Click = ButtonAction(Button.Press)
```

Comment:
> Can you pick one? Press or Click.

Fare enough! Let me do it in a followup patch after this cascade of changes will land so I don't have to rebase all of the pulls based on this just for this.